great_expectations/agent/actions/test_draft_datasource_config.py

```diff
@@ -0,0 +1,49 @@
+from uuid import UUID
+
+import pydantic
+
+from great_expectations.agent.actions import ActionResult, AgentAction
+from great_expectations.agent.agent import GxAgentEnvVars
+from great_expectations.agent.models import DraftDatasourceConfig
+from great_expectations.core.http import create_session
+
+
+class DraftDatasourceConfigAction(AgentAction[DraftDatasourceConfig]):
+    def run(self, event: DraftDatasourceConfig, id: str) -> ActionResult:
+        draft_config = self.get_draft_config(config_id=event.config_id)
+        datasource_type = draft_config.get("type", None)
+        if datasource_type is None:
+            raise ValueError(
+                "The DraftDatasourceConfigAction can only be used with a fluent-style datasource."
+            )
+        try:
+            datasource_cls = self._context.sources.type_lookup[datasource_type]
+        except KeyError as exc:
+            raise ValueError(
+                "DraftDatasourceConfigAction received an unknown datasource type."
+            ) from exc
+        datasource = datasource_cls(**draft_config)
+        datasource.test_connection(
+            test_assets=True
+        )  # raises exception if config isn't valid
```

Comment:
This comment is misleading.
Exceptions will be raised on `datasource_cls(**draft_config)` if the config is invalid according to pydantic model validation.

`test_connection()` can still raise exceptions even if the config is "valid", there could be a number of reasons why a connection failed. A temporary outage, expired credentials, slow internet leading to a timeout etc.

Reply:
sure - [493ef06](https://github.com/great-expectations/great_expectations/pull/8410/commits/493ef0663ac7233d36c4b686b114e17540e6eeeb)